persist/fs/commitlog/bitset_test.go

```diff
@@ -0,0 +1,176 @@
+			}),
+		InitialPreConditionFunc: func(state commands.State) bool {
+			return true
+		},
+		GenCommandFunc: func(state commands.State) gopter.Gen {
+			return gen.OneGenOf(genBitSetTestCommand,
+				genBitSetSetCommand, genBitSetClearAllCommand)
+		},
+	})
+	properties.Property("BitSet set/test/clearAll", prop)
+	properties.TestingRun(t)
+}
+
+type bitSetTestState struct {
+	set     *bitSet
+	currSet map[uint]struct{}
+}
+
+var genBitSetTestCommand = gen.
+	UIntRange(0, bitSetPropTestMaxValueLength).
+	Map(func(v uint) commands.Command {
+		return &commands.ProtoCommand{
+			Name: "Test",
+			PreConditionFunc: func(state commands.State) bool {
+				return true
+			},
+			RunFunc: func(q commands.SystemUnderTest) commands.Result {
+				state := q.(bitSetTestState)
+				return state.set.test(v)
+			},
+			NextStateFunc: func(state commands.State) commands.State {
+				return state
+			},
+			PostConditionFunc: func(state commands.State, result commands.Result) *gopter.PropResult {
+				actual := result.(bool)
+				_, expected := state.(bitSetTestState).currSet[v]
+				if actual == expected {
+					return &gopter.PropResult{Status: gopter.PropTrue}
+				}
+				return &gopter.PropResult{
+					Status: gopter.PropFalse,
+					Error:  fmt.Errorf("expected %d to be true when testing value", v),
+				}
+			},
+		}
+	})
+
+var genBitSetSetCommand = gen.
+	UIntRange(0, bitSetPropTestMaxValueLength).
+	Map(func(v uint) commands.Command {
+		return &commands.ProtoCommand{
+			Name: "Set",
+			PreConditionFunc: func(state commands.State) bool {
+				return true
+			},
+			RunFunc: func(q commands.SystemUnderTest) commands.Result {
+				state := q.(bitSetTestState)
+				state.set.set(v)
+				if !state.set.test(v) {
+					return fmt.Errorf("not set, expected set after setting %v", v)
+				}
+				return nil
+			},
+			NextStateFunc: func(state commands.State) commands.State {
+				s := state.(bitSetTestState)
+				s.currSet[v] = struct{}{}
+				return s
+			},
+			PostConditionFunc: func(state commands.State, result commands.Result) *gopter.PropResult {
+				if result == nil {
+					return &gopter.PropResult{Status: gopter.PropTrue}
+				}
+				return &gopter.PropResult{
+					Status: gopter.PropFalse,
+					Error:  result.(error),
+				}
+			},
+		}
+	})
+
+var genBitSetClearAllCommand = gen.Const(&commands.ProtoCommand{
+	Name: "ClearAll",
+	PreConditionFunc: func(state commands.State) bool {
+		return true
+	},
+	RunFunc: func(q commands.SystemUnderTest) commands.Result {
+		s := q.(bitSetTestState)
+		s.set.clearAll()
+		return nil
+	},
+	NextStateFunc: func(state commands.State) commands.State {
+		s := state.(bitSetTestState)
+		for k := range s.currSet {
+			delete(s.currSet, k)
+		}
+		return s
+	},
+	PostConditionFunc: func(state commands.State, result commands.Result) *gopter.PropResult {
+		s := state.(bitSetTestState)
+		for i := range s.set.values {
+			if s.set.values[i] != 0 {
+				return &gopter.PropResult{
+					Status: gopter.PropFalse,
```

Comment:
I'd suggest using `gopter.PropError`. With `PropFalse` `gopter` will fail but won't log the error